feat(gux-datepicker): added disabled prop

reginald-cui-genesys opened the pull request:
COMUI-720
PDNA-13344

Added disabled prop to gux-datepicker, and updated the unit tests

katie-bobbe-genesys commented:
I think it might make sense to keep the labels for the `gux-datepicker` at 100% opacity. That way it will match the styling of the `gux-form-field` components.
![image](https://user-images.githubusercontent.com/82100934/141861669-2bfdddf2-fa22-40ed-9cfe-f907a5a79b11.png)

![image](https://user-images.githubusercontent.com/82100934/141861580-89d98272-7887-48c3-9548-ad811acab44d.png)

reginald-cui-genesys commented:
Addressed comment and changed it so the label is always at 100% opacit
![Screen Shot 2021-11-15 at 2 49 04 PM](https://user-images.githubusercontent.com/77636863/141865040-55d17191-3d45-4463-b54a-bcd3e6b3588e.png)

daragh-king-genesys commented on src/components/stable/gux-datepicker/gux-datepicker.tsx:
```diff
@@ -659,13 +666,18 @@ export class GuxDatepicker {
+          class={`gux-datepicker-field-input ${
+            this.disabled ? 'gux-disabled' : ''
+          }`}
+        >
```
I don't think you need to add `gux-disabled` here you can reference the `gux-disabled` on the root element

Something like
```
.gux-datepicker {
    &.gux-disabled {
        pointer-events: none;
        cursor: default;

        .gux-datepicker-field {
            opacity: 0.5;
        }
    }
}
```

reginald-cui-genesys replied:
Changes made

daragh-king-genesys commented on src/components/stable/gux-datepicker/gux-datepicker.tsx:
```diff
@@ -685,13 +697,18 @@ export class GuxDatepicker {
+          class={`gux-datepicker-field-input ${
+            this.disabled ? 'gux-disabled' : ''
+          }`}
+        >
```
Same
